File: src-tauri/system_info/src/models/drive_model.rs
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize)]
#[serde(rename_all = "PascalCase")]
pub struct DriveModel {
    pub name: String,
    pub label: Option<String>,
    pub total_space: u64,
    pub available_space: u64,
}

File: src/app/core/models/drive-model.ts
export interface DriveModel{
    Name:string,
    Label:string,
    TotalSpace:number,
    AvailableSpace:number,
}

export function defaultDriveModel():DriveModel{
    return {
        Name:"C:",
        Label:"C",
        TotalSpace:100,
        AvailableSpace:100
    }
}